### PR TITLE
docs: replace CommonJS with ESM

### PR DESCRIPTION
**Pull request by tigt**
Obsoletes #1808 

## Description

- Not updating the example in `10-awesome-marko-features.md` because that example clearly dates itself by mentioning it’s 2017

- Deleted `server-side-rendering.md` because it wasn’t in the navigation (only a couple backlinks even exposed it), and it was redundant with the Rendering page

## Questions

- A couple places (`marko-vs-react.md`, `why-is-marko-fast.md`) show a Marko template’s rendered output. Should I update it to the uglier, longer outputs we have now? Verbatim or rewritten to be easier to follow?

- ~~Do we even want the `server-side-rendering.md#Bootstrapping: Non-Lasso` section anymore? Is it only for people trying to integrate Marko with a new/unsupported bundler?~~ Obviated by deleting `server-side-rendering.md`

## Checklist:

- [x] I have read the **CONTRIBUTING** document and have signed (or will sign) the CLA.
- [x] I have updated/added documentation affected by my changes.
- ~~[ ] I have added tests to cover my changes.~~

### CODE REVIEW

File: packages/marko/docs/marko-vs-react.md [hunk 1]
```diff
@@ -617,20 +617,22 @@ style.less {
 
 ### API
 
-Marko compiles component to JavaScript modules that export an API for rendering
-the component as shown below:
+Marko compiles components to JavaScript modules with exported API for rendering,
```
Comment (DylanPiercey) [inline packages/marko/docs/marko-vs-react.md:620]: I don't think this is grammatically correct

File: packages/marko/docs/marko-vs-react.md [hunk 2]
```diff
@@ -617,20 +617,22 @@ style.less {
...
+as shown below:
 
 ```js
-require("./components/greeting")
+import greetingComponent from "./components/greeting.marko";
```
Comment (DylanPiercey) [inline packages/marko/docs/marko-vs-react.md:624]: Maybe just condense to the following?

```js
import Greeting
```

Comment (tigt) [reply packages/marko/docs/marko-vs-react.md:624]: Just here, or do you want to move to `PascalCase` for `.marko` imports across the docs from now on?

File: packages/marko/docs/marko-vs-react.md (hunk 1 above)
Comment (tigt) [reply packages/marko/docs/marko-vs-react.md]: How about the new version?

File: packages/marko/docs/marko-vs-react.md (hunk 2 above)
Comment (DylanPiercey) [reply packages/marko/docs/marko-vs-react.md:624]: In think either `Template` or the component name in pascal case throughout the docs would be good.

Comment (tigt) [reply packages/marko/docs/marko-vs-react.md:624]: I’ll change the instance here to keep this ticket scoped, but we should probably follow up with @mlrawlings to get a Project entry for that particular consistency across the rest of the docs

**Pull request merged**